### ISSUE TITLE
[Bug] Cannot setup integration

### ISSUE THREAD

**Issue by fcastilloec** [Vaskivskyi/ha-asusrouter#152]
### The problem

I can't configure this integration. I thought it was related to the default port, but after reading the additional info in the README, and realizing that I have to force an error before I'm allowed to set a custom port number. Now I'm getting a weird error and can't pass the configuration stage.

### Your device model

RT-AC68U

### Firmware type

Merlin

### Firmware version

386.7

### Integration version with the issue

0.5.1

### Method of the integration installation

HACS

### What version and type of Home Assistant installation do you use

2022.7.0

### Is there anything useful in the logs?

```txt
This error originated from a custom integration.

Logger: aiohttp.server
Source: custom_components/asusrouter/config_flow.py:374
Integration: asusrouter (documentation, issues)
First occurred: 5:45:51 PM (3 occurrences)
Last logged: 5:48:18 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 79, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 136, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 177, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 62, in wrapper
    result = await method(view, request, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 109, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 277, in async_configure
    result = await self._async_handle_step(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 359, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/asusrouter/config_flow.py", line 481, in async_step_device
    return await self.async_select_step(step_id, errors)
  File "/config/custom_components/asusrouter/config_flow.py", line 374, in async_select_step
    return await self._steps[f"{last_step}_error"](errors=errors)
KeyError: 'device_error'

```

**Comment by Vaskivskyi** [comment]
Hi, @fcastilloec,

Thank you for your feedback! 👍🏻 

I am sorry, that the integration doesn't work for you. I will look into this issue and let you know the approximate timeframe for the fix. But, let's say, a couple of days.

The idea of separating port and other specific settings from credentials is to simplify the setup for most of the users. You don't have to "force error" to be able to select a port. This should happen by itself. I will look into this as well.

**Comment by fcastilloec** [comment]
I've updated my Home Assistant a couple of times since this issue was reported, currently running (2022.7.3), HACS has also been updated (which I don't think is as relevant) to version 1.26.0. I'm not able to set up your integration without any issues. I needed it primarily for presence tracking and it's working perfectly.
I'm not sure if this issue should be closed yet, but I can't replicate this error anymore.
